### ISSUE TITLE
Issue: File paths are unquoted

### ISSUE THREAD

**Issue by joebobmiles** [nektos/act#748]
## System information

 - Operating System: Windows 10
 - Architecture: x64 (64-bit)
 - Apple M1: no
 - Docker version: 20.10.6
 - `act` version: 0.2.23

## Expected behavior

Running `act`, with no parameters, should run all tasks without accessing a non-existent file path that results from the truncation of a path that contains spaces.

## Actual behavior

`act` fails to quote file paths when sending commands to docker. This results in a file path that contains spaces (which is common on Windows computers) being truncated by the command line. Docker then attempts to use the truncated path, resulting in an error that causes workflows to fail.

## Workflow and/or repository

I made a test repository that [replicates the problem](https://github.com/joebobmiles/didactic-octo-barnacle).

### From scratch:

1. Create a directory that contains a space in the name, such as "Hello World".
    ```bash
    $ # Linux or MacOS
    $ mkdir "Hello World"
    ```
    ```pwsh
    > # Windows Powershell or CMD
    > md "Hello World"
    ```
2. Under that directory, create a `.github/workflows` directory.
    ```bash
    $ # Linux or MacOS
    $ cd "Hello World"
    $ mkdir -p .github/workflows
    ```
    ```pwsh
    > # Windows Powershell or CMD
    > cd "Hello World"
    > md .github\workflows
    ```
4. In `.github\workflows`, create a workflow file that contains the following:
    ```yml
    name: Test
    on: push
    
    jobs:
      test:
        runs-on: ubuntu-latest
        steps:
          - run: sudo apt-get update
    ```
3. Run `act` with no arguments.

### From repo:

1. Clone [the reproduction repository](https://github.com/joebobmiles/didactic-octo-barnacle).
2. Move into the `Hello World` directory and run `act` with no arguments.

## `act` output

<details>
  <summary>Log</summary>

```none
time="2021-07-01T17:07:50-07:00" level=debug msg="Loading environment from C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\Hello World\\.env"
time="2021-07-01T17:07:50-07:00" level=debug msg="Loading secrets from C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\Hello World\\.secrets"
time="2021-07-01T17:07:50-07:00" level=debug msg="Loading workflows from 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\Hello World\\.github\\workflows'"
time="2021-07-01T17:07:50-07:00" level=debug msg="Loading workflows recursively"
time="2021-07-01T17:07:50-07:00" level=debug msg="Found workflow 'test.yml' in 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\Hello World\\.github\\workflows\\test.yml'"
time="2021-07-01T17:07:50-07:00" level=debug msg="Reading workflow 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\Hello World\\.github\\workflows\\test.yml'"
time="2021-07-01T17:07:50-07:00" level=debug msg="Correcting if statements 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\Hello World\\.github\\workflows\\test.yml'"
time="2021-07-01T17:07:50-07:00" level=debug msg="Planning event: push"
time="2021-07-01T17:07:50-07:00" level=debug msg="Loading slug from git directory 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\.git'"
time="2021-07-01T17:07:50-07:00" level=debug msg="Found revision: f451cde05e6d93b59afdb98d6e0f67859dcaa313\n"
time="2021-07-01T17:07:50-07:00" level=debug msg="Loading revision from git directory 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\.git'"
time="2021-07-01T17:07:50-07:00" level=debug msg="Found revision: f451cde05e6d93b59afdb98d6e0f67859dcaa313\n"
time="2021-07-01T17:07:50-07:00" level=debug msg="HEAD points to 'f451cde05e6d93b59afdb98d6e0f67859dcaa313'"
time="2021-07-01T17:07:50-07:00" level=debug msg="using github ref: refs/heads/master"
time="2021-07-01T17:07:50-07:00" level=debug msg="context env => map[ACT:true]"
[Test/test] 🚀  Start image=catthehacker/ubuntu:act-latest
time="2021-07-01T17:07:50-07:00" level=debug msg="Loading slug from git directory 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\.git'"
time="2021-07-01T17:07:50-07:00" level=debug msg="Found revision: f451cde05e6d93b59afdb98d6e0f67859dcaa313\n"
time="2021-07-01T17:07:50-07:00" level=debug msg="Loading revision from git directory 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\.git'"
time="2021-07-01T17:07:50-07:00" level=debug msg="Found revision: f451cde05e6d93b59afdb98d6e0f67859dcaa313\n"
time="2021-07-01T17:07:50-07:00" level=debug msg="HEAD points to 'f451cde05e6d93b59afdb98d6e0f67859dcaa313'"
time="2021-07-01T17:07:50-07:00" level=debug msg="using github ref: refs/heads/master"
[Test/test]   🐳  docker pull catthehacker/ubuntu:act-latest
time="2021-07-01T17:07:50-07:00" level=debug msg="Image exists? true"
[Test/test] Removed container: a190b1424d9bf64ec8c19e5e0816c3d10d0725e4ddae99efa6fd3fcd73548c34
[Test/test]   🐳  docker volume rm act-Test-test
[Test/test]   🐳  docker create image=catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Test/test] Created container name=act-Test-test id=f9e73c7f4e902d7372bf8efa4da58f4ad4336997def5e03ba276257063ca53c6 from image catthehacker/ubuntu:act-latest (platform: )
[Test/test] ENV ==> [RUNNER_TOOL_CACHE=/opt/hostedtoolcache RUNNER_OS=Linux RUNNER_TEMP=/tmp]
[Test/test]   🐳  docker run image=catthehacker/ubuntu:act-latest platform= entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Test/test] Starting container: f9e73c7f4e902d7372bf8efa4da58f4ad4336997def5e03ba276257063ca53c6
[Test/test] Started container: f9e73c7f4e902d7372bf8efa4da58f4ad4336997def5e03ba276257063ca53c6
[Test/test]   🐳  docker exec cmd=[mkdir -m 0777 -p /var/run/act] user=root
[Test/test] Exec command '[mkdir -m 0777 -p /var/run/act]'
time="2021-07-01T17:07:51-07:00" level=debug msg="Writing entry to tarball workflow/event.json len:2"
time="2021-07-01T17:07:51-07:00" level=debug msg="Writing entry to tarball workflow/envs.txt len:0"
time="2021-07-01T17:07:51-07:00" level=debug msg="Writing entry to tarball workflow/paths.txt len:0"
[Test/test] Extracting content to '/var/run/act/'
time="2021-07-01T17:07:51-07:00" level=debug msg="Loading slug from git directory 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\.git'"
time="2021-07-01T17:07:51-07:00" level=debug msg="Found revision: f451cde05e6d93b59afdb98d6e0f67859dcaa313\n"
time="2021-07-01T17:07:51-07:00" level=debug msg="Loading revision from git directory 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\.git'"
time="2021-07-01T17:07:51-07:00" level=debug msg="Found revision: f451cde05e6d93b59afdb98d6e0f67859dcaa313\n"
time="2021-07-01T17:07:51-07:00" level=debug msg="HEAD points to 'f451cde05e6d93b59afdb98d6e0f67859dcaa313'"
time="2021-07-01T17:07:51-07:00" level=debug msg="using github ref: refs/heads/master"
time="2021-07-01T17:07:51-07:00" level=debug msg="Loading slug from git directory 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\.git'"
time="2021-07-01T17:07:51-07:00" level=debug msg="Found revision: f451cde05e6d93b59afdb98d6e0f67859dcaa313\n"
time="2021-07-01T17:07:51-07:00" level=debug msg="Loading revision from git directory 'C:\\Users\\josep\\Documents\\Programming\\act-unquoted-file-paths\\.git'"
time="2021-07-01T17:07:51-07:00" level=debug msg="Found revision: f451cde05e6d93b59afdb98d6e0f67859dcaa313\n"
time="2021-07-01T17:07:51-07:00" level=debug msg="HEAD points to 'f451cde05e6d93b59afdb98d6e0f67859dcaa313'"
time="2021-07-01T17:07:51-07:00" level=debug msg="using github ref: refs/heads/master"
time="2021-07-01T17:07:51-07:00" level=debug msg="context env => map[ACT:true AGENT_TOOLSDIRECTORY:/opt/hostedtoolcache DEPLOYMENT_BASEPATH:/opt/runner IMAGE_OS:ubuntu20 ImageOS:ubuntu20 LSB_RELEASE:20.04 PATH:/opt/hostedtoolcache/node/14.17.1/x64/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin RUNNER_USER:root RUN_TOOL_CACHE:/opt/hostedtoolcache USER:root]"
time="2021-07-01T17:07:51-07:00" level=debug msg="context env => map[ACT:true AGENT_TOOLSDIRECTORY:/opt/hostedtoolcache CI:true DEPLOYMENT_BASEPATH:/opt/runner GITHUB_ACTION:0 GITHUB_ACTIONS:true GITHUB_ACTION_REF: GITHUB_ACTION_REPOSITORY: GITHUB_ACTOR:nektos/act GITHUB_API_URL:https://api.github.com GITHUB_BASE_REF: GITHUB_ENV:/var/run/act/workflow/envs.txt GITHUB_EVENT_NAME:push GITHUB_EVENT_PATH:/var/run/act/workflow/event.json GITHUB_GRAPHQL_URL:https://api.github.com/graphql GITHUB_HEAD_REF: GITHUB_JOB:test GITHUB_PATH:/var/run/act/workflow/paths.txt GITHUB_REF:refs/heads/master GITHUB_REPOSITORY:joebobmiles/didactic-octo-barnacle GITHUB_REPOSITORY_OWNER:joebobmiles GITHUB_RETENTION_DAYS:0 GITHUB_RUN_ID:1 GITHUB_RUN_NUMBER:1 GITHUB_SERVER_URL:https://github.com GITHUB_SHA:f451cde05e6d93b59afdb98d6e0f67859dcaa313 GITHUB_TOKEN: GITHUB_WORKFLOW:Test GITHUB_WORKSPACE:/mnt/c/Users/josep/Documents/Programming/act-unquoted-file-paths/Hello World IMAGE_OS:ubuntu20 ImageOS:ubuntu20 LSB_RELEASE:20.04 PATH:/opt/hostedtoolcache/node/14.17.1/x64/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin RUNNER_PERFLOG:/dev/null RUNNER_TRACKING_ID: RUNNER_USER:root RUN_TOOL_CACHE:/opt/hostedtoolcache USER:root]"
[Test/test] setupEnv => map[ACT:true AGENT_TOOLSDIRECTORY:/opt/hostedtoolcache CI:true DEPLOYMENT_BASEPATH:/opt/runner GITHUB_ACTION:0 GITHUB_ACTIONS:true GITHUB_ACTION_REF: GITHUB_ACTION_REPOSITORY: GITHUB_ACTOR:nektos/act GITHUB_API_URL:https://api.github.com GITHUB_BASE_REF: GITHUB_ENV:/var/run/act/workflow/envs.txt GITHUB_EVENT_NAME:push GITHUB_EVENT_PATH:/var/run/act/workflow/event.json GITHUB_GRAPHQL_URL:https://api.github.com/graphql GITHUB_HEAD_REF: GITHUB_JOB:test GITHUB_PATH:/var/run/act/workflow/paths.txt GITHUB_REF:refs/heads/master GITHUB_REPOSITORY:joebobmiles/didactic-octo-barnacle GITHUB_REPOSITORY_OWNER:joebobmiles GITHUB_RETENTION_DAYS:0 GITHUB_RUN_ID:1 GITHUB_RUN_NUMBER:1 GITHUB_SERVER_URL:https://github.com GITHUB_SHA:f451cde05e6d93b59afdb98d6e0f67859dcaa313 GITHUB_TOKEN: GITHUB_WORKFLOW:Test GITHUB_WORKSPACE:/mnt/c/Users/josep/Documents/Programming/act-unquoted-file-paths/Hello World IMAGE_OS:ubuntu20 ImageOS:ubuntu20 LSB_RELEASE:20.04 PATH:/opt/hostedtoolcache/node/14.17.1/x64/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin RUNNER_PERFLOG:/dev/null RUNNER_TRACKING_ID: RUNNER_USER:root RUN_TOOL_CACHE:/opt/hostedtoolcache USER:root]
[Test/test] ⭐  Run sudo apt-get update
time="2021-07-01T17:07:51-07:00" level=debug msg="Wrote command '\nsudo apt-get update\n' to 'workflow/0'"
time="2021-07-01T17:07:51-07:00" level=debug msg="Writing entry to tarball workflow/0 len:19"
[Test/test] Extracting content to '/mnt/c/Users/josep/Documents/Programming/act-unquoted-file-paths/Hello World'
[Test/test]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /mnt/c/Users/josep/Documents/Programming/act-unquoted-file-paths/Hello World/workflow/0] user=
[Test/test] Exec command '[bash --noprofile --norc -e -o pipefail /mnt/c/Users/josep/Documents/Programming/act-unquoted-file-paths/Hello World/workflow/0]'
| bash: /mnt/c/Users/josep/Documents/Programming/act-unquoted-file-paths/Hello: No such file or directory
[Test/test]   ❌  Failure - sudo apt-get update
time="2021-07-01T17:07:51-07:00" level=debug msg="exit with `FAILURE`: 1"
Error: exit with `FAILURE`: 1
```

</details>

**Comment by nikakaltura** [comment]
This is a serious issue. I can't run any bach commands, because my user has a space in the name

**Comment by BaileyJM02** [comment]
I am also having this issue, prefixing whitespace with a backward slash should fix it if quotes break some commands.

**Comment by catthehacker** [comment]
The problem was with whole command being re-split after building it. Will be fixed by #772
